### PR TITLE
feat(vertexai): Add stream text sample

### PR DESCRIPTION
**Pull request by ymotongpoo**
## Description

Go porting of the following Python sample: 
https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/generative_ai/inference/stream_text_basic.py

Region tag is `generativeaionvertexai_stream_text_basic`

## Checklist
- [x] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [x] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [x] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [x] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Review by grayside — CHANGES_REQUESTED** [review]
Thanks for the sample, I've asked for a small change to unblock approval. The feedback is identical to what I left on #4148

File: vertexai/stream-text-basic/streamtextbasic.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+	iter := model.GenerateContentStream(
+		ctx,
+		genai.Text("Write a story about a magic backpack."),
+	)
+	for {
+		resp, err := iter.Next()
+		if err == iterator.Done {
+			break
+		}
```
[the remark this answers is not in the document]
Comment (ymotongpoo) [reply vertexai/stream-text-basic/streamtextbasic.go:48]: ACK. I added the case for empty response. same as #4148.

File: vertexai/stream-text-basic/streamtextbasic.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+// [END generativeaionvertexai_stream_text_basic]
+
+func printResponse(w io.Writer, resp *genai.GenerateContentResponse) {
+	fmt.Fprint(w, "generated response: ")
+	for _, c := range resp.Candidates {
+		for _, p := range c.Content.Parts {
+			fmt.Fprintf(w, "%s ", p)
+		}
+	}
+}
```
[the remark this answers is not in the document]
Comment (ymotongpoo) [reply vertexai/stream-text-basic/streamtextbasic.go:66]: ACK. Inlined the custom print function content into the main function.

**Review by grayside — APPROVED** [review]
LGTM, thank you!

**Pull request merged**